pkg/controllers/servicescale/scaler.go

```diff
@@ -305,11 +292,32 @@ func (s *SimpleScale) scrape() error {
 	}
 	stat.readyPods = readyPods
 
-	logrus.Debugf("collect metric for %s/%s, total request: %v, average in-flight request per pod: %v, ready pod: %v", s.namespace, s.serviceName, totalActiveRequest, stat.activeRequest, readyPods)
+	logrus.Debugf("collect metric for %s/%s, total request: %v(inbound %v, outbound %v), average in-flight request per pod: %v, ready pod: %v", s.namespace, s.serviceName, totalActiveRequest, inboundActiveRequests, outbountActiveRequests, stat.activeRequest, readyPods)
 	s.metrics.stats = append(s.metrics.stats, stat)
 	return nil
 }
 
+func calculateActiveRequests(scanner *bufio.Scanner, requestMatch, responseMatch string, direction string) int {
+	var request, response int
+	for scanner.Scan() {
+		if strings.Contains(scanner.Text(), responseMatch) && strings.Contains(scanner.Text(), direction) {
+			parts := strings.Split(scanner.Text(), " ")
+			if len(parts) == 2 {
+				rqs, _ := strconv.Atoi(parts[1])
+				response = rqs
+			}
+		}
+		if strings.Contains(scanner.Text(), requestMatch) && strings.Contains(scanner.Text(), direction) {
+			parts := strings.Split(scanner.Text(), " ")
+			if len(parts) == 2 {
+				rqs, _ := strconv.Atoi(parts[1])
+				request = rqs
+			}
+		}
```

Comment:
Not very dry, that or I can't spot the difference. Can we abstract into method and pass in direction.